python-avd/pyavd/_eos_designs/shared_utils/routing.py

```diff
@@ -99,6 +99,29 @@ def isis_instance_name(self: SharedUtilsProtocol) -> str | None:
         # This point cannot be reached because the function won't be called if either of the conditions in the if-block is not satisfied.
         return None
 
+    @cached_property
+    def bgp_as_notation(self: SharedUtilsProtocol) -> Literal["asdot", "asplain"]:
+        bgp_as_notation = self.inputs.bgp_as_notation
+        if bgp_as_notation == "asdot" or (bgp_as_notation == "auto" and "." in str(self.bgp_as)):
+            return "asdot"
+        return "asplain"
+
+    def get_asn(self: SharedUtilsProtocol, asn: str | int | None) -> str | None:
+        if asn is None:
+            return None
+
+        bgp_as_notation = self.bgp_as_notation
+        if bgp_as_notation == "asdot" and "." not in str(asn):
+            if (prefix := int(asn) // 65536) != 0:
+                return f"{prefix}.{int(asn) % 65536}"
+            return f"{int(asn) % 65536}"
+
+        if bgp_as_notation == "asplain" and "." in str(asn):
+            prefix, suffix = map(int, str(asn).split("."))
+            return str(int(prefix) * 65536 + int(suffix))
```

Comment:
same can we not return some very bad value here (this probably should be handled in schema)

if I do `bgp_as: -12345.12234`

Reply:
so the following (bad input I know)

```
    - name: RD-RT-ADMIN-SUBFIELD-L3LEAF4
      bgp_as: -12345.123-127
      id: 4
```

in asplain will render

```
-router bgp 65004
+router bgp -809041794
    router-id 192.168.255.4
    update wait-install
    no bgp default ipv4-unicast
```

but maybe we can ignore this for now.

@ClausHolbechArista

Reply:
let's add a TODO in the code for this @Vibhu-gslab that will be "protected" from once we have schema support for format (soon!)